### PR TITLE
Ports material and makeshift armor from Polaris

### PR DESCRIPTION
**Pull request by Alberyk**
-adds support to material clothing, already used by rings and makeshift weapon
-adds makeshift armor that you can craft using buckets for helmets and material sheets to make the plates
-adds a wooden bucket that can be crafted using planks also ported from polaris
-adds icon_has_variants to some sheets, adding sprites to them from tg
-fixes bronze being just a place holder
-adds sheet types to bronze and titanium

plasteelarmor

### CODE REVIEW

File: code/modules/clothing/clothing.dm [hunk 1]
```diff
@@ -103,6 +122,124 @@
...
+	material = get_material_by_name(new_material)
+	if(!material)
+		qdel(src)
+	else
+		name = "[material.display_name] [initial(name)]"
+		health = round(material.integrity/10)
+		if(applies_material_color)
+			color = material.icon_colour
+		if(material.products_need_process())
+			STOP_PROCESSING(SSprocessing, src)
```
Comment (Lohikar) [inline code/modules/clothing/clothing.dm]: Shouldn't this be `START_PROCESSING`?

Comment (Alberyk) [reply code/modules/clothing/clothing.dm]: Fixed.

File: code/modules/clothing/clothing.dm [hunk 2]
```diff
@@ -103,6 +122,124 @@
...
+/obj/item/clothing/proc/set_material(var/new_material)
+	material = get_material_by_name(new_material)
+	if(!material)
+		qdel(src)
+	else
+		name = "[material.display_name] [initial(name)]"
+		health = round(material.integrity/10)
+		if(applies_material_color)
+			color = material.icon_colour
+		if(material.products_need_process())
```
Comment (skull132) [inline code/modules/clothing/clothing.dm]: I'm not seeing an actual process() override implemented in this PR? What does this bit even doooo.

Comment (Alberyk) [reply code/modules/clothing/clothing.dm]: It is used by some materials, such as uranium.

**Pull request merged**